omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -587,10 +582,27 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
                     self._table.numberOfPages = Math.floor(initialRowCount/self._table.fnSettings()._iDisplayLength)+1;
 
             } else if(self.options.searchPhrase || self.options.doSearchWhenEmpty) {
-                if (self.options.searchPhrase == null) {
-                    self.options.searchPhrase = "";
-                }
-                $j(input).val(self.options.searchPhrase).keyup();
+            	if (self.options.searchPhrase == null) {
+            		self.options.searchPhrase = "";
+            	}                
+            	//check to see if we need to select the checkbox for includeVerbose and/or includeVoided
+            	//since there is a searchPhrase the page will reload using these values
+            	if(self.options.showVerbose){
+            		$j('#includeVerbose').attr('checked','checked');
+            	}
+            	if(self.options.includeVoided){
+            		$j('#includeVoided').attr('checked','checked');
+            	}
+            	$j(input).val(self.options.searchPhrase).keyup();
+            	//setting the cursor to the end of the searchPhrase
+            	//setSelectionRange is a method of the DOM element not the jQuery object so get the DOM element
+            	var inputNodeDOM = $j('#inputNode').get(0);
+            	if(inputNodeDOM){
+            		var searchPhraseLength = self.options.searchPhrase.length;
+            		if (inputNodeDOM.setSelectionRange) {
+            			inputNodeDOM.setSelectionRange(searchPhraseLength, searchPhraseLength);
```

Comment:
This code is what sets the cursor to the end of the text for firefox and for chrome. I was not able to find a way to test on IE. I have a check though that if selectionRange is not browser supported it will ignore it which is what should happen for IE. There is createRange for IE i believe but I did not want to add code I could not see the behavior for since I could not track down a copy of IE ( I only run Ubuntu). I can verify that it works with Firefox and Chrome however. Would you rather I remove it still? If you would like for it to stay I do need to add a new commit. I accidentally lost the code when I commited for getting the focus for the chrome browser.